extension/EdFi.Ods.Extensions.LMSX/LMS-Harmonizer/1090-lms-assignment-submissions.sql

```diff
@@ -0,0 +1,67 @@
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+CREATE OR ALTER PROCEDURE [lms].[harmonize_assignment_submissions] AS
+BEGIN
+    SET NOCOUNT ON;
+	select * from edfi.Descriptor edfides
+	inner join lmsx.SubmissionStatusDescriptor lmsdes on edfides.DescriptorId = lmsdes.SubmissionStatusDescriptorId
+
+	SELECT
+		lmsSubmission.AssignmentSubmissionIdentifier, --[AssignmentSubmissionIdentifier],
+		lmsUser.EdFiStudentId, --[StudentUSI],
+		lmsxAssignment.AssignmentIdentifier, --[AssignmentIdentifier],
+		lmsxAssignment.SchoolId, --[SchoolId],
+		submissionStatusDescriptor.DescriptorId, --[SubmissionStatusDescriptorId],
+		lmsSubmission.SubmissionDateTime, --[SubmissionDateTime],
+		lmsSubmission.EarnedPoints, --[EarnedPoints],
+		lmsSubmission.Grade, --[Grade],
+		lmsSubmission.CreateDate,
+		lmsSubmission.LastModifiedDate,
+		lmsSubmission.DeletedAt
+	INTO #ALL_SUBMISSIONS
+	FROM LMS.AssignmentSubmission lmsSubmission
+	INNER JOIN LMSX.Assignment lmsxAssignment
+		ON lmsSubmission.AssignmentIdentifier = lmsxAssignment.AssignmentIdentifier
+	INNER JOIN LMS.LMSUser lmsUser
+		ON lmsUser.LMSUserIdentifier = lmsSubmission.LMSUserIdentifier
+	INNER JOIN EDFI.Descriptor submissionStatusDescriptor
+		ON submissionStatusDescriptor.CodeValue = lmsSubmission.SubmissionStatus
+
+		INNER JOIN LMSX.SubmissionStatusDescriptor lmsxSubmissionStatus
+			ON submissionStatusDescriptor.DescriptorId = lmsxSubmissionStatus.SubmissionStatusDescriptorId
```

Comment:
At some point a system will have tens of thousands of assignments. That is not a large number in terms of relational databases, but it does make me wonder... should this query explicitly exclude any records that have already been uploaded, so that the temp table is not too large?

Oh, but then you would need separate queries for update and delete.

Let's keep this. We can optimize later if needed.